lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -36,8 +41,24 @@ def execute(message)
               job_msg = job ? "#{job.id}[#{job.class_name}]" : 'unknown job'
               @logger.info "Error processing job #{job_msg}: #{e}"
               @logger.debug e.backtrace.join("\n")
+            ensure
+              poison_pill = true
             end
           end
+
+          @monitor.post(message) do |message|
+            while poison_pill
+              begin
+                # Extend the visibility timeout to the provided refresh timeout
+                message.change_visibility({ visibility_timeout: @refresh_timeout })
+                # Wait half the refresh timeout, and repeat
+                sleep(@refresh_timeout / 2)
+              rescue => e
+                # If anything goes wrong, we want to handle it. We don't care what.
```

Comment:
We may want to not log certain issues - like if the message has already been deleted and the change_visibility call fails.  Haven't thought through all of the error cases, but at least that one seems expected.

Reply:
I can leave this out and just drop out for now?

Reply:
Added the catch for that. I don't see any other exceptions that we should treat specially?